mautic/mautic
# [DPMMA-2888] Correct change field type Form API

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
When editing a form via API, when we want to change the field type, e.g. select to string, an error appears. Regardless of whether we use the PUT or PATCH method.

Message: The response has unexpected status code (400).Response: {"errors":[{"code":400,"message":"properties: This form should not contain extra fields.","details":[]}]}

Endpoint PATCH /forms/ID/edit







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. change the form structure by changing the field type, e.g. from select to string, use the RESTAPI PATCH method /forms/ID/edit
